lib/active_merchant/billing/gateways/braintree/braintree_common.rb

```diff
@@ -26,6 +26,9 @@ def scrub(transcript)
 
   def supports_network_tokenization?
     response = authorize(1, ActiveMerchant::Billing::NetworkTokenizationCreditCard.test_credit_card)
+    response_code = response.params['braintree_transaction'].try!(:[], 'processor_response_code')
+    raise(ActiveMerchant::InvalidResponseError, response.message) unless response_code
```

Comment:
So, was this actually necessary?

My original questions were: 

> What happens in case of a connection error or 5xy response and this body isn't there? It is appropriate that we let that bubble up (probably, yes)? I'm assuming a request error will be raised before we get to the hash traversal, yeah?

None of which were really answered.

This is also the incorrect error to raise here. It's only ever used elsewhere in Active Merchant in case of accept-encoding errors, or actually malformed HTTP.